lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
We'll need to update docs: https://docs.puppet.com/puppet/4.10/lang_reserved.html#tags